### PR TITLE
Feature whatsapp logic connection

### PR DESCRIPTION
**Pull request by VincePaulin**
Texts are only in English now (and for the time being).
I've prepared a showDialog for double-factor authetification that can be used for each social network.

whatsapp_dialog
whatsapp_response
whatsapp_success

### CODE REVIEW

File: assets/l10n/intl_en.arb [hunk 1]
```diff
@@ -2550,5 +2550,49 @@
...
+  "youDonTHaveConversation": "You don't have any conversation yet.",
+  "connectChatNetworks": "Connect your chat networks",
+  "addSocialMessagingAccounts": "Add your social messaging accounts",
+  "addSocialMessagingAccountsText": "Connect all your conversations!",
+  "connected": "Connected",
+  "notConnected": "Not connected",
+  "connectYourSocialAccount": "Login to your account",
+  "enterYourDetails": "Enter your details :",
+  "pleaseEnterPassword": "Please enter your password",
+    "usernameNotFound": "The username you entered doesn't appear to belong to an account. Please check your username and try again.",
```
Comment (ignyx) [inline assets/l10n/intl_en.arb]: why is the indentation not the same ?

File: lib/pages/chat_list/chat_list_view.dart [hunk 2]
```diff
@@ -86,26 +86,33 @@ class ChatListView extends StatelessWidget {
...
+                      : ActiveFilter.allChats),
+          onPopInvoked: (pop) async {
+            if (pop) return;
             final selMode = controller.selectMode;
             if (controller.isSearchMode) {
               controller.cancelSearch();
-              return false;
+              return;
             }
             if (selMode != SelectMode.normal) {
```
Comment (ignyx) [inline lib/pages/chat_list/chat_list_view.dart]: use `else if`s (applicable to multiple nearby lines)

File: lib/pages/add_bridge/qr_code_connect.dart [hunk 3]
```diff
@@ -0,0 +1,239 @@
...
+
+  @override
+  void initState() {
+    super.initState();
+
+    responseFuture = widget.botConnection.fetchDataWhatsApp();
+  }
+
+  @override
+  Widget build(BuildContext context) {
```
Comment (ignyx) [inline lib/pages/add_bridge/qr_code_connect.dart:46]: function too long, some elements could be move to separate functions/variables

File: lib/pages/add_bridge/error_message_dialog.dart [hunk 4]
```diff
@@ -0,0 +1,109 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_gen/gen_l10n/l10n.dart';
+
+// Display a showDialog with an error message related to the identifier
+void showErrorUsernameDialog(BuildContext context) {
```
Comment (ignyx) [inline lib/pages/add_bridge/error_message_dialog.dart]: There's a lot of duplicate code in this file. I feel like some refactoring could simplify (and shorten) this file

File: lib/pages/add_bridge/connection_bridge_dialog.dart [hunk 5]
```diff
@@ -0,0 +1,320 @@
...
+            actions: <Widget>[
+              TextButton(
+                onPressed: () {
+                  Navigator.of(context).pop();
+                  completer.complete(false);
+                },
+                child: Text(L10n.of(context)!.cancel),
+              ),
+              TextButton(
+                onPressed: () async {
```
Comment (ignyx) [inline lib/pages/add_bridge/connection_bridge_dialog.dart:286]: could be move to separate function/component

File: lib/pages/add_bridge/connection_bridge_dialog.dart [hunk 6]
```diff
@@ -0,0 +1,320 @@
...
+          child: AlertDialog(
+            title: Text(
+              "${L10n.of(context)!.connectYourSocialAccount} ${network.name}",
+              style: const TextStyle(
+                fontSize: 20.0,
+                fontWeight: FontWeight.bold,
+                // color: Color(0xFFFAAB22),
+              ),
+            ),
+            content: Form(
```
Comment (ignyx) [inline lib/pages/add_bridge/connection_bridge_dialog.dart]: could probably be move to separate function/component

**Review by ignyx — CHANGES_REQUESTED** [review]
Very nice ! Some refactoring could help lighten things up

**Comment by ignyx** [comment]
Additionally, merging `dev` into this branch would :
- resolve conflicts the other way around
- make the difference more easy to read

File: assets/l10n/intl_en.arb (hunk 1 above)
Comment (VincePaulin) [reply assets/l10n/intl_en.arb]: The formatting doesn't work for arb, so I'll get into the habit of arranging it by hand

File: lib/pages/chat_list/chat_list_view.dart (hunk 2 above)
Comment (VincePaulin) [reply lib/pages/chat_list/chat_list_view.dart]: Fluffychat modified it in today's merge dev

**Comment by ignyx** [comment]
Looks great ! One small detail : after confirming the login on the phone (whatsapp app), nothing is updated in the Tawkie app. Monitoring login would add complexity. Easiest thing to do would be to add a text saying roughly "After logging in you can return to messages" and maybe a button navigating to messages

**Comment by VincePaulin** [comment]
I've tested it several times and always get a message when the connection is established or when the time has run out.
We could look at it together and see how we could do things differently.

In the meantime, I've improved the response listening function so that it stops when the user leaves the page

**Pull request merged**